src/emulator/storage/rules/manager.ts

```diff
@@ -100,13 +120,50 @@ export class StorageRulesManager {
   }
 }
 
-function readSourceFile(fileName: string): string {
-  try {
-    return fs.readFileSync(fileName).toString();
-  } catch (error: any) {
-    if (error.code === "ENOENT") {
-      throw new FirebaseError(`File not found: ${fileName}`);
+/**
+ * Maintains a mapping from storage resource to {@link StorageRulesManagerImplementation} and
+ * directs calls to the appropriate instance.
+ */
+class StorageRulesManagerRegistry implements StorageRulesManager {
+  private _rulesManagers: Map<string, StorageRulesManagerImplementation>;
```

Comment:
Initialize _rulesManagers right here instead of in the constructor:

```ts
private _rulesManagers = new Map<string, StorageRulesManagerImplementation>();
```